includes/OrderController.php

```diff
@@ -47,22 +47,53 @@ public function __construct(SettingsRepository $settings, OrderRepository $order
         $this->orderRepository = $orderRepository;
 
         \add_action('rest_api_init', function () {
-            register_rest_route(self::NAMESPACE, '/update/(?P<id>\d+)', [
+            register_rest_route(self::NAMESPACE, '/orders/(?P<id>\d+)/callbacks/success', [
                 'methods' => 'POST',
-                'callback' => [$this, 'handleOrderUpdateRequest'],
+                'callback' => [$this, 'handleSuccessCallback'],
+                'permission_callback' => '__return_true'
+            ]);
+
+            register_rest_route(self::NAMESPACE, '/orders/(?P<id>\d+)/callbacks/fulfillment', [
+                'methods' => 'POST',
+                'callback' => [$this, 'handleFulfillmentCallback'],
                 'permission_callback' => '__return_true'
             ]);
         });
     }
 
     /**
-     * The callback handler.
+     * The callback handler for success callbacks.
      *
      * @param WP_REST_Request $request the post request.
      */
-    public function handleOrderUpdateRequest(WP_REST_Request $request): WP_REST_Response
+    public function handleSuccessCallback(WP_REST_Request $request): WP_REST_Response
     {
         $id = $request->get_params()['id'];
+        $data = json_decode($request->get_body());
+
+        $order = $this->orderRepository->findById($id);
+
+        if (is_null($order)) {
+            return new WP_REST_Response(null, 404, null);
+        }
+
+        $order->update_meta_data($this->settings->getOrderIdFieldName(), $data->orderId);
+
+        $order->save();
+
+        return new WP_REST_Response(null, 200, null);
+    }
+
+    /**
+     * The callback handler for fulfillment callbacks.
+     *
+     * @param WP_REST_Request $request The post request.
+     * @return WP_REST_Response
+     */
+    public function handleFulfillmentCallback(WP_REST_Request $request): WP_REST_Response
+    {
+        $id = $request->get_params()['id'];
```

Comment:
i assume the id will be definitely in the request params

Reply:
The ID is in the URL, hence it has to be there.